.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,89 @@
+name: e2e
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build-and-run-example:
+    runs-on: ubuntu-latest
+    environment: restricted
+    steps:
+    - name: Create resource suffix
+      run: >
+        echo "SUFFIX=$(echo $RANDOM | md5sum | head -c6)"
```

Comment:
Is there a way to make this `SUFFIX` look like `<pr number>-<random number>`, just for convenience. If the cleanup fails and we had to do manual clean up.

Reply:
> Re: Static Infrastructure: Is it just permissions and a RG?

yes, also the `restricted` env needs client_id, tenant_id, subscription_id set as secrets and on the azure side there is an app + federated credentials + role assignments.